actors/market/src/state.rs

```diff
@@ -8,23 +8,59 @@ use cid::Cid;
 use fil_actors_runtime::runtime::Policy;
 use fil_actors_runtime::{
     actor_error, make_empty_map, make_map_with_root_and_bitwidth, ActorDowncast, ActorError, Array,
-    AsActorError, Map, Set, SetMultimap,
+    Set, SetMultimap,
 };
 use fvm_ipld_blockstore::Blockstore;
 use fvm_ipld_encoding::tuple::*;
 use fvm_ipld_encoding::Cbor;
+use fvm_ipld_hamt::BytesKey;
 use fvm_shared::address::Address;
 use fvm_shared::clock::{ChainEpoch, EPOCH_UNDEFINED};
 use fvm_shared::deal::DealID;
 use fvm_shared::econ::TokenAmount;
 use fvm_shared::error::ExitCode;
 use fvm_shared::HAMT_BIT_WIDTH;
 use num_traits::Zero;
+use std::collections::BTreeMap;
 
 use super::policy::*;
 use super::types::*;
 use super::{DealProposal, DealState};
 
+fn deal_get_payment_remaining(
```

Comment:
Please move helper functions to the bottom of the file

Reply:
Done 👍🏾